app/com/linkedin/drelephant/schedulers/OozieScheduler.java

```diff
@@ -0,0 +1,187 @@
+  public static final String OOZIE_JOB_DEF_URL_TEMPLATE = "oozie_job_url_template";
+  public static final String OOZIE_JOB_EXEC_URL_TEMPLATE = "oozie_job_exec_url_template";
+  public static final String OOZIE_WORKFLOW_DEF_URL_TEMPLATE = "oozie_workflow_url_template";
+  public static final String OOZIE_WORKFLOW_EXEC_URL_TEMPLATE = "oozie_workflow_exec_url_template";
+
+  private String schedulerName;
+  private String jobName;
+  private String jobExecId;
+  private String workflowExecId;
+  private String workflowName;
+  private int workflowDepth;
+
+  private OozieClient oozieClient;
+  private String jobDefUrlTemplate;
+  private String jobExecUrlTemplate;
+  private String workflowDefUrlTemplate;
+  private String workflowExecUrlTemplate;
+
+  public OozieScheduler(String appId, Properties properties, SchedulerConfigurationData schedulerConfData) {
+    this(appId, properties, schedulerConfData, null);
+  }
+  public OozieScheduler(String appId, Properties properties, SchedulerConfigurationData schedulerConfData, OozieClient oozieClient) {
+    schedulerName = schedulerConfData.getSchedulerName();
+
+    if (properties != null && properties.getProperty(OOZIE_ACTION_ID) != null) {
+      this.oozieClient = oozieClient == null ? makeOozieClient(schedulerConfData) : oozieClient;
+      jobDefUrlTemplate = schedulerConfData.getParamMap().get(OOZIE_JOB_DEF_URL_TEMPLATE);
+      jobExecUrlTemplate = schedulerConfData.getParamMap().get(OOZIE_JOB_EXEC_URL_TEMPLATE);
+      workflowDefUrlTemplate = schedulerConfData.getParamMap().get(OOZIE_WORKFLOW_DEF_URL_TEMPLATE);
+      workflowExecUrlTemplate = schedulerConfData.getParamMap().get(OOZIE_WORKFLOW_EXEC_URL_TEMPLATE);
+
+      loadInfo(appId, properties);
+    } else {
+      // Use default value of data type
+    }
+  }
+
+  private void loadInfo(String appId, Properties properties) {
+    // 0004167-160629080632562-oozie-oozi-W@some-action
+    jobExecId = properties.getProperty(OOZIE_ACTION_ID);
+    if (jobExecId.contains("@")) {
+      String[] pair = jobExecId.split("@", 2);
+      workflowExecId = pair[0];
+      jobName = pair[1];
+
+      try {
+        logger.info("Fetching Oozie workflow info for " + workflowExecId);
+        WorkflowJob workflow = oozieClient.getJobInfo(workflowExecId);
+        logger.info("Oozie workflow for " + workflowExecId + ": " + workflow);
+        workflowName = workflow.getAppName();
+      } catch (OozieClientException e) {
+        throw new RuntimeException("Failed fetching Oozie workflow " + workflowExecId + " info", e);
+      }
+    }
+
+    workflowDepth = 0; // TODO: Add sub-workflow support
+  }
+
+  private OozieClient makeOozieClient(SchedulerConfigurationData schedulerConfData) {
+    String oozieApiUrl = schedulerConfData.getParamMap().get(OOZIE_API_URL);
+
+    if (oozieApiUrl == null) {
+      throw new RuntimeException("Missing " + OOZIE_API_URL + " param for Oozie Scheduler");
+    }
+
+    return new OozieClient(oozieApiUrl);
```

Comment:
I believe OozieClient doesn't work when kerberos is enabled. AuthOozieClient might be a better option